### DISCUSSION TITLE
Recommended approach to using `flutter_map` with BLoC architecture?

### DISCUSSION THREAD

**Discussion by comatory** [fleaflet/flutter_map#948] [Q&A]
I was wondering if anyone here is using BLoC pattern and this plugin. I'm using `flutter_bloc` plugin and I was hoping to move all my logic inside a map BLoC and have all the logic inside the BLoC.
It's a type of architecture that I'm trying to achieve where widgets are really just UI most of the time and rest of the logic is extracted.

I think I could create a BLoC and register `controller` that can be passed to `MapOptions` and achieve lot of the stuff there. If anyone has advice or examples of how they implement this, it 'd be great to hear about them.

**Comment by themisir** [comment]
I think this question is opinion based. It really depends on the project that what's considered business logic and what's UI logic.

**Reply by comatory** [reply]
Yes definitely. Since this is a discussion I wouldn't mind "opinionated" examples.

**Reply by ibrierley** [reply]
If you want an opinion :D, I'd say if you are at a stage of asking the question, Bloc is probably not the right one to use. I hated it, and it simply got in the way of what I wanted to do. Look into that when more familiar with state management and you have proper separation of business logic. I'm not convinced 90% of the things most people do here need this, but I'm happy to be convinced otherwise.

Otherwise I would look into riverpod, getx etc, or simply at first have a singleton class or something (at first).

Main thing I think is not to get bogged down with state at first, you can lose a lot of time there, and spend time in the wrong areas. But do consider a separation of concerns in your architecture, then it's easier to adapt later as well.
